cluster/addons/kube-proxy/kube-proxy-ds.yaml

```diff
@@ -34,7 +34,7 @@ spec:
         effect: "NoSchedule"
       containers:
       - name: kube-proxy
-        image: {{pillar['kube_docker_registry']}}/kube-proxy-amd64:{{pillar['kube-proxy_docker_tag']}}
+        image: {{pillar['kube_docker_registry']}}/kube-proxy-{{pillar['host_arch']}}:{{pillar['kube-proxy_docker_tag']}}
```

Comment:
Can we use docker multi-arch manifest list to avoid specifying arch at all?

Reply:
This will require changing upstream release process to produce multi-arch image first. We can follow up on that separately to decouple these changes, WDYT?

Reply:
Sounds good. Let's create a bug to track.

Reply:
SG. Created https://github.com/kubernetes/kubernetes/issues/99245 and added TODO. I squashed the commits to be ready to be merged. Please LGTM again. Thank you!